compiler/damlc/daml-stdlib-src/DA/Foldable.daml

```diff
@@ -54,39 +51,35 @@ class Foldable t where
 
   -- | List of elements of a structure, from left to right.
   toList : t a -> [a]
-  toList t = foldl (\b a -> a::b) [] t
+  toList t = foldr (::) [] t
 
   -- | Test whether the structure is empty. The default implementation is optimized for structures that are similar to cons-lists, because there is no general way to do better.
   null : t a -> Bool
-  null = foldl (\_ _ -> False) True
+  null = foldr (\_ _ -> False) True
```

Comment:
I assume `foldr` is more efficient.

Reply:
Good, I've also been assuming that :slightly_smiling_face:

Reply:
I think neither is, since in either case we traverse the entire structure - no laziness

```haskell
module Main2 where

import Prelude hiding (foldl, foldr)
import DA.Monoid (Any (..))
import Daml.Script
import DA.Foldable

main : Script ()
main = script do
  let list = [1,2,3,4]

  debugRaw "\nnull by foldr"
  debug $ nullFoldR list

{-
null by foldr -- the list is reversed because I'm using the bad `DA.Foldable.foldr`, but the point is that the entire structure is traversed
  "nullFoldR.f (4) (True)"
  "nullFoldR.f (3) (False)"
  "nullFoldR.f (2) (False)"
  "nullFoldR.f (1) (False)"
  False
-}

  debugRaw "\nnull by foldMap"
  debug $ nullFoldMap list

{-
null by foldMap
  "nullFoldMap.f (1)"
  "nullFoldMap.f (2)"
  "nullFoldMap.f (3)"
  "nullFoldMap.f (4)"
  False
-}

nullFoldR : (Foldable f, Show a) => f a -> Bool
nullFoldR =
  foldr
    (\x y -> trace ("nullFoldR.f (" <> show x <> ") (" <> show y <> ")") False)
    True

nullFoldMap : (Foldable f, Show a) => f a -> Bool
nullFoldMap =
  getAny . foldMap
    (\x -> trace ("nullFoldMap.f (" <> show x <> ")") Any False)
```

Reply:
I assume foldr is often more efficient that foldl as it is by default implemented using foldr. While foldr should be defined by user or use foldMap defined by the user. 
It is definitively not always true.